syntax = "proto3";
import "proto/response.proto";

enum RequestType {
  Image = 0;
  Components = 1;
  Details = 2;
  Video = 3;
}

message Request {
  string node_name = 1;
  RequestType request_type = 2;
}
 
message Response {
  string node_name = 1;
  string description = 2;

  oneof DetailedResponse {
    NoneResponse none_response = 3;
    ImageResponse image_response = 4;
  }
}
